packages/components/organisms/f-cookie-banner/src/components/CookieBanner.vue

```diff
@@ -0,0 +1,364 @@
+         * @param {String} cookieValue
+         */
+        setCookieBannerCookie (cookieValue) {
+            this.$cookies.set('je-cookieConsent', cookieValue, {
+                path: '/',
+                maxAge: this.cookieExpiry
+            });
+        },
+        /**
+         * Set the legacy cookie banner cookie
+         */
+        setLegacyCookieBannerCookie () {
+            this.$cookies.set('je-banner_cookie', '2', {
+                path: '/',
+                maxAge: this.cookieExpiry
+            });
+        },
+        /**
+         * Push tracking events
+         * @param {String} consentLevel
+         */
+        dataLayerPush (consentLevel) {
+            const dataLayer = window.dataLayer || [];
+            dataLayer.push({ event: 'trackConsent', userData: { consent: consentLevel } });
+        },
+        /**
+         * Check for excluded cookies/storage
+         * @param {String} cookieName
+         * @returns {Bool}
+         */
+        isNotExcluded (cookieName) {
+            return this.config.cookieExclusionList.every(arrVal => cookieName.lastIndexOf(arrVal, arrVal.length - 1) === -1);
+        },
+        /**
+         * Remove unnecessary cookies
+         */
+        removeUnnecessaryCookies () {
+            const cookies = Object.keys(this.$cookies.getAll());
+            for (let i = 0; i < cookies.length; i++) {
+                if (this.isNotExcluded(cookies[i])) this.$cookies.remove(cookies[i]);
+            }
+        },
+        /**
+         * Check for iOS browser
+         * @returns {Bool}
+         */
+        checkIosSafari () {
+            return /(iPhone|iPad).*Safari/.test(navigator.userAgent);
+        },
+        /**
+         * Resend GTM events
+         */
+        resendEvents () {
+            const dataLayer = window.dataLayer || [];
+            const excludeEvents = ['gtm.js', 'gtm.click', 'gtm.linkClick', 'gtm.triggerGroup', 'trackConsent', 'showConsent'];
+            let i;
+            for (i = 0; i < dataLayer.length; i++) {
+                if (dataLayer[i].event && !excludeEvents.includes(dataLayer[i].event)) {
+                    dataLayer[i].eventResent = dataLayer[i].event;
+                    dataLayer[i]['gtm.uniqueEventId'] = Math.floor(100000 + Math.random() * 900000);
+                    dataLayer.push(dataLayer[i]);
+                }
+                if (dataLayer[i].event && dataLayer[i].event === 'trackConsent' && (dataLayer[i].userData.consent === 'full' || dataLayer[i].userData.consent === 'necessary')) {
+                    dataLayer.push({ eventResent: false });
+                    break;
+                }
+            }
+        }
+    }
+};
+</script>
+
+<style lang="scss" module>
+    .c-cookieBanner {
+        font-family: $font-family-base;
+    }
+
+    .c-cookieBanner-overlay {
+        position: fixed;
+        top: 0;
+        bottom: 0;
+        left: 0;
+        right: 0;
+        background-color: rgba(0, 0, 0, 0.7);
```

Comment:
We should have a suitable background colour in our fozzie-colour-palette

Reply:
@dandel10n I'll raise this as a separate ticket to add a modal background colour to the fozzie-color-palette, as we'll need to update the backgrounds for all the components that use modals and this work is rolling on way too long now.